bioblend/galaxy/datasets/__init__.py

```diff
@@ -104,7 +104,8 @@ def download_dataset(self, dataset_id, file_path=None, use_default_filename=True
 
         # Don't use self.gi.make_get_request as currently the download API does
         # not require a key
-        r = requests.get(url, verify=self.gi.verify)
+        stream_content = file_path is not None
+        r = requests.get(url, verify=self.gi.verify, stream=stream_content)
```

Comment:
I just noticed that we were missing a `r.raise_for_status()` here, can you please add it?